node-manager/src/socket.rs

```diff
@@ -0,0 +1,413 @@
+        let peer_manager_copy = self.peer_manager.clone();
+        let connected_copy = self.connected.clone();
+        debug!("spawning multi socket connection reader");
+        spawn_local(async move {
+            while let Some(msg) = conn_copy.read.lock().await.next().await {
+                if let Ok(Message::Bytes(msg)) = msg {
+                    // parse the msg and see which pubkey it belongs to
+                    if msg.len() < PUBKEY_BYTES_LEN {
+                        debug!("msg not long enough to have pubkey, ignoring...");
+                        continue;
+                    }
+                    let (id_bytes, message_bytes) = msg.split_at(PUBKEY_BYTES_LEN);
+
+                    // now send that data to the right subsocket
+                    let socket_lock = socket_map_copy.lock().await;
+                    let found_subsocket = socket_lock.get(id_bytes);
+                    match found_subsocket {
+                        Some(s) => {
+                            match s.send(Message::Bytes(message_bytes.to_vec())) {
+                                Ok(_) => (),
+                                Err(e) => error!("error sending msg to channel: {}", e),
+                            };
+                        }
+                        None => {
+                            drop(socket_lock);
+
+                            // create a new subsocket and pass it to peer_manager
+                            debug!(
+                                "no connection found for socket address, creating new: {:?}",
+                                id_bytes
+                            );
+                            let inbound_subsocket = WsSocketDescriptor::Mutiny(
+                                create_new_subsocket(
+                                    socket_map_copy.clone(),
+                                    send_to_multi_socket_copy.clone(),
+                                    id_bytes.to_vec(),
+                                )
+                                .await,
+                            );
+                            debug!("created new subsocket: {:?}", id_bytes);
+                            match peer_manager_copy
+                                .new_inbound_connection(inbound_subsocket.clone(), None)
+                            {
+                                Ok(_) => {
+                                    debug!("gave new subsocket to peer manager: {:?}", id_bytes);
+                                    schedule_descriptor_read(
+                                        inbound_subsocket,
+                                        peer_manager_copy.clone(),
+                                    );
+
+                                    // now that we have the inbound connection, send the original
+                                    // message to our new subsocket descriptor
+                                    match socket_map_copy.lock().await.get(id_bytes) {
```

Comment:
could we have `new_inbound_connection` return the subsocket so we don't need this lookup